### ISSUE TITLE
cooler version 0.8.11 incompatible with numpy 1.24

### ISSUE THREAD

**Issue by lldelisle** [open2c/cooler#303]
Hi,
I got an error:
```
>>> import cooler
/home/ldelisle/.conda/envs/test2/lib/python3.8/site-packages/cooler/util.py:690: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.  (This may have returned Python scalars in past versions.
  columns, dtype=None, index_columns=None, index_names=None, default_dtype=np.object
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ldelisle/.conda/envs/test2/lib/python3.8/site-packages/cooler/__init__.py", line 14, in <module>
    from .api import Cooler, annotate
  File "/home/ldelisle/.conda/envs/test2/lib/python3.8/site-packages/cooler/api.py", line 22, in <module>
    from .util import parse_cooler_uri, parse_region, open_hdf5, closing_hdf5
  File "/home/ldelisle/.conda/envs/test2/lib/python3.8/site-packages/cooler/util.py", line 690, in <module>
    columns, dtype=None, index_columns=None, index_names=None, default_dtype=np.object
  File "/home/ldelisle/.conda/envs/test2/lib/python3.8/site-packages/numpy/__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'object'
>>> 
```
This error seems to be linked to numpy 1.24 as `import cooler` works if I use numpy 1.23.
Should I update the conda recipe?

**Comment by lldelisle** [comment]
I wrote a PR to update the bioconda recipe: https://github.com/bioconda/bioconda-recipes/pull/38816

**Comment by golobor** [comment]
@nvictus , couldn't the solution be to use np.object_ instead?..

**Comment by nvictus** [comment]
>@nvictus , couldn't the solution be to use np.object_ instead?

Yes, @robomics has already made the PR and I merged it earlier this week.

v3.9 will be coming out shortly and this will be fixed. In the meantime, updating the conda recipe for 0.8.11 or 0.8.* makes sense.

> Should I update the conda recipe?

Yes. Thank you, Lucille!

**Comment by nvictus** [comment]
Fixed by db9478a
